libs/server/StoreWrapper.cs

```diff
@@ -413,6 +414,44 @@ async Task CompactionTask(int compactionFrequencySecs, CancellationToken token =
             }
         }
 
+        async Task HashCollectTask(int hashCollectFrequencySecs, CancellationToken token = default)
+        {
+            Debug.Assert(hashCollectFrequencySecs > 0);
+            try
+            {
+                var scratchBufferManager = new ScratchBufferManager();
+                using var storageSession = new StorageSession(this, scratchBufferManager, null, null, logger);
+                var key = ArgSlice.FromPinnedSpan("*"u8);
+
+                while (true)
+                {
+                    if (token.IsCancellationRequested) return;
+
+                    if (objectStore is null)
+                    {
+                        logger?.LogWarning("HashCollectFrequencySecs option is configured but Object store is disabled. Stopping the background hash collect task.");
+                        return;
+                    }
+
+                    var header = new RespInputHeader(GarnetObjectType.Hash) { HashOp = HashOperation.HCOLLECT };
+                    var input = new ObjectInput(header);
+
+                    storageSession.HashCollect(key, ref input, ref storageSession.objectStoreBasicContext);
+                    scratchBufferManager.Reset();
+
+                    await Task.Delay(hashCollectFrequencySecs * 1000, token);
```

Comment:
Maybe use a TimeSpan here
```suggestion
                    await Task.Delay(TimeSpan.FromSeconds(hashCollectFrequencySecs), token);
```

Reply:
Done